### PR TITLE
Feat image gallery upload

### PR DESCRIPTION
**Pull request by Amartyakumar**
Added the image upload functionality

### CODE REVIEW

File: pages/apps/_app/event/_id/index.vue [hunk 1]
```diff
@@ -714,7 +870,68 @@ export default {
...
+      this.formData.Other.push(data[0])
+      this.updateEventGallery(this.formData)
+    },
+    async updateEventGallery(formData) {
+      const res = await this.$axios
+        .$patch(
+          `https://${nuxtconfig.axios.eventUrl}${nuxtconfig.axios.apiEndpoint}Events/${this.$route.params.id}`,
+          formData
+        )
+        .catch((e) => console.log('Error', e))
```
Comment (SukantGujar) [inline pages/apps/_app/event/_id/index.vue:930]: Don't mix promises with async

**Review by SukantGujar — CHANGES_REQUESTED** [review]

**Review by SukantGujar — APPROVED** [review]

File: components/common/copy/index.vue [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    onCopy() {
+      const writeBtn = document.querySelector(`.${this.uniqueId}btn`)
+      const inputEl = document.querySelector(`.${this.uniqueId}`)
+
+      writeBtn.addEventListener('click', () => {
+        const inputValue = inputEl.value.trim()
+        if (inputValue) {
+          navigator.clipboard
+            .writeText(inputValue)
+            .then(() => {
```
Comment (SukantGujar) [inline components/common/copy/index.vue:46]: No promise, use async await.

File: components/common/copy/index.vue [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+      snackbarText: 'Copied to clipboard',
+      timeout: 1000,
+    }
+  },
+  methods: {
+    onCopy() {
+      const writeBtn = document.querySelector(`.${this.uniqueId}btn`)
+      const inputEl = document.querySelector(`.${this.uniqueId}`)
+
+      writeBtn.addEventListener('click', () => {
```
Comment (SukantGujar) [inline components/common/copy/index.vue:41]: Why is this listener required? This code will keep on adding listener on each onCopy method call. You need to remove the listener once the copy is over.

File: pages/apps/_app/event/_id/index.vue [hunk 4]
```diff
@@ -728,7 +942,118 @@ export default {
...
+      try {
+        formData.map(async (id) => {
+          const res = await this.$axios.$put(
+            `https://${nuxtconfig.axios.eventUrl}${nuxtconfig.axios.apiEndpoint}Events/${this.$route.params.id}/Others/rel/${id}`
+          )
+          setTimeout(() => {
+            this.snackbarText = 'Attachment added successfully'
+            this.snackbar = true
+            this.refresh()
+          }, 5000)
```
Comment (SukantGujar) [inline pages/apps/_app/event/_id/index.vue:1050]: Why a timeout of 5 seconds here?

File: pages/apps/_app/event/_id/index.vue [hunk 5]
```diff
@@ -757,6 +1082,45 @@ export default {
       const regUrl = `https://${nuxtconfig.axios.eventUrl}/e/${this.data.event.UniqLink}`
       window.open(`${regUrl}`, '_blank')
     },
+    async deleteBannerFile(e, id) {
+      const checkRes = confirm('Are you sure you want to delete')
+      if (checkRes) {
+        const res = await this.$axios.delete(
+          `https://${nuxtconfig.axios.eventUrl}${nuxtconfig.axios.apiEndpoint}Events/${this.$route.params.id}/BannerImage/${id}`
+        )
+        if (res) {
```
Comment (SukantGujar) [inline pages/apps/_app/event/_id/index.vue:1091]: There's no setTimeout used here. Why is it used earlier?

**Review by SukantGujar — CHANGES_REQUESTED** [review]

File: pages/apps/_app/event/_id/index.vue (hunk 5 above)
Comment (Amartyakumar) [reply pages/apps/_app/event/_id/index.vue:1090]: resolved

File: pages/apps/_app/event/_id/index.vue (hunk 4 above)
Comment (Amartyakumar) [reply pages/apps/_app/event/_id/index.vue]: resolved

File: components/common/copy/index.vue (hunk 3 above)
Comment (Amartyakumar) [reply components/common/copy/index.vue]: resolved

File: components/common/copy/index.vue (hunk 2 above)
Comment (Amartyakumar) [reply components/common/copy/index.vue]: resolved

**Review by SukantGujar — APPROVED** [review]

**Pull request merged**